data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -49,45 +76,189 @@ def prepare_data(self, protocol_name: str, path: str = LOCAL_STORAGE_PATH) -> di
+
+    @classmethod
+    def _get_response(cls, protocol_name: str) -> requests.Response:
+        """
+        Gets the response from endpoint for the given protocol.
+        :param protocol_name: str
+        :return: requests.Response
+        """
+        response = requests.get(cls.INTEREST_MODEL_VALUES_URL.format(protocol=protocol_name))
+        response.raise_for_status()
+        return response.json()
+
+    @classmethod
+    def get_interest_rate_models_data(cls, protocol_name: str) -> dict:
+        """
+        Returns an interest rate models data
+        :param protocol_name: str
+        :return: dict
+        """
+        cls.verify_endpoint_protocol(protocol_name)
+        for protocol in AvailableProtocolID:
+            current_protocol = str(protocol.value).lower()
+            if protocol_name.lower() == current_protocol:
+                try:
+                    return cls._get_response(protocol.value)
+                except:
+                    time.sleep(10)
+                    return cls._get_response(protocol.value)
+
+    @staticmethod
+    def normalize_protocol_name(protocol_name: str) -> str:
+        """
+        Normalizes the given protocol name.
+        :param protocol_name: str
+        :return: str
+        """
+        return "_".join(protocol_name.split(" ")).lower()
+
+    def _calculate_liquidable_debt(
+            self, data: dict = None,
+    ) -> dict:
         """
         Calculates liquidable debt based on data provided and updates an existing data.
-        :param data: Data to calculate liquidable debt for.
+        Data to calculate liquidable debt for:
+        :param data: {
+            Row number: {
+                Field name: Token value
+                 or Decimal - Field value (the price as example)
+                 or {
+                    Token name: Token value
+                }
+            }
+        }
         :return: A dictionary of the ready liquidable debt data.
         """
-        result_data = deepcopy(data)
+        # FIXME Read loan_entities from the database
+        # FIXME concentrate for one Protocol and one collateral_token and debt_token
+        # FIXME go through all data and init them as loan_entities
+        result_data = dict()
+        current_prices = Prices()
+        asyncio.run(current_prices.get_lp_token_prices())
+        INTEREST_RATE_MODELS_MAPPING = {
+            protocol.value.lower(): self.get_interest_rate_models_data(protocol_name=protocol.value)
+            for protocol in AvailableProtocolID
+        }
 
-        for row_number in data:
-            state = self.state_class(verbose_user=data[row_number][USER_FIELD_NAME])
+        # init just once for improving performance
+        state = self.state_class()
+        # TODO: 1. Read data from db for state.loan_entities from LoanState model
+        # TODO: 2. Set last interest rate record by block
+        # state.interest_rate_models = InterestRateModels(..)
```

Comment:
No, `collateral_interest_rate_models` and `debt_interest_rate_models` are enough. No need for a third attribute.